### DISCUSSION TITLE
Make Swagger example field empty dynamically

### DISCUSSION THREAD

**Discussion by piankma** [fastapi/fastapi#11815] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Description

In my use-case, for specific environments we want to keep Swagger for quick API calls. But - users being users, don't mind thinking and click like monkeys. So we end up with bunch of fields with "string" as a content.

Is there a quick way to overwrite all the examples, that are being generated into swagger file - dynamically?

**Comment by jgould22** [comment, marked as answer]
You can edit the resulting openapi spec, though I am not sure I would describe it as "quick"

https://fastapi.tiangolo.com/how-to/extending-openapi/#extending-openapi

**Comment by piankma** [comment]
For now I came up with this:

```py
class ExampleModel(pydantic.BaseModel):
    model_config: ConfigDict = {
        "json_schema_extra": {
            "examples": ({} if True else None)
        }
    }
    ...
```

But a global solution would be appreciated. Leaving this open till then.

**Comment by zheng1** [comment]
Please have a look at this to see if it can help you
```python
def json_schema_extra(schema: JsonDict, model: type[BaseModel]):
    print(schema)
    for name, prop in schema["properties"].items():
        if name == "kind":
            prop["examples"] = [model.__name__]
    return schema


class Model(BaseModel):
    kind: str = Field(examples=["Foo"])
    model_config = ConfigDict(json_schema_extra=json_schema_extra)

class Pagination(Model):
    limit: int
    total_page: int
```
